pandas/core/internals/blocks.py

```diff
@@ -599,14 +599,13 @@ def _astype(self, dtype, copy=False, errors='raise', values=None,
                 return self.copy()
             return self
 
-        if klass is None:
-            if is_sparse(self.values):
-                # special case sparse, Series[Sparse].astype(object) is sparse
-                klass = ExtensionBlock
-            elif is_object_dtype(dtype):
-                klass = ObjectBlock
-            elif is_extension_array_dtype(dtype):
-                klass = ExtensionBlock
+        if is_sparse(self.values):
+            # special case sparse, Series[Sparse].astype(object) is sparse
+            klass = ExtensionBlock
+        elif is_object_dtype(dtype):
+            klass = ObjectBlock
+        elif is_extension_array_dtype(dtype):
+            klass = ExtensionBlock
```

Comment:
someone more familiar with this than me may want to double-check if these 7 lines are still needed.

Reply:
yes @TomAugspurger recently modified this for Sparse